app/drivers/kscan/kscan_gpio_matrix.c

```diff
@@ -51,257 +158,300 @@ static int kscan_gpio_config_interrupts(const struct device **devices,
-                COND_CODE_0(DT_INST_PROP(n, debounce_period), ({ k_work_submit(&data->work); }),   \
-                            ({                                                                     \
-                                k_delayed_work_cancel(&data->work);                                \
-                                k_delayed_work_submit(&data->work, K_MSEC(5));                     \
-                            }))                                                                    \
-            } else { kscan_gpio_enable_interrupts_##n(dev); })                                     \
-        return 0;                                                                                  \
-    }                                                                                              \
-    static void kscan_gpio_work_handler_##n(struct k_work *work) {                                 \
-        struct kscan_gpio_data_##n *data = CONTAINER_OF(work, struct kscan_gpio_data_##n, work);   \
-        kscan_gpio_read_##n(data->dev);                                                            \
-    }                                                                                              \
-    COND_INTERRUPTS(static void kscan_gpio_irq_callback_handler_##n(                               \
-        const struct device *dev, struct gpio_callback *cb, gpio_port_pins_t pin) {                \
-        struct kscan_gpio_irq_callback_##n *data =                                                 \
-            CONTAINER_OF(cb, struct kscan_gpio_irq_callback_##n, callback);                        \
-        kscan_gpio_disable_interrupts_##n(data->dev);                                              \
-        COND_CODE_0(DT_INST_PROP(n, debounce_period), ({ k_work_submit(data->work); }), ({         \
-                        k_delayed_work_cancel(data->work);                                         \
-                        k_delayed_work_submit(data->work,                                          \
-                                              K_MSEC(DT_INST_PROP(n, debounce_period)));           \
-                    }))                                                                            \
-    })                                                                                             \
+    // While interrupts are enabled, set all outputs active so a pressed key
+    // will trigger an interrupt.
+    return kscan_matrix_set_all_outputs(dev, 1);
+}
+#endif
+
+#if USE_INTERRUPTS
+static int kscan_matrix_interrupt_disable(const struct device *dev) {
+    int err = kscan_matrix_interrupt_configure(dev, GPIO_INT_DISABLE);
+    if (err) {
+        return err;
+    }
+
+    // While interrupts are disabled, set all outputs inactive so
+    // kscan_matrix_read() can scan them one by one.
+    return kscan_matrix_set_all_outputs(dev, 0);
+}
+#endif
+
+#if USE_INTERRUPTS
```

Comment:
We're already in an `#if USE_INTERRUPTS` block. Any reason you've chosen to duplicate the preprocessor blocks around each function?

Reply:
IMO it makes it easier to tell that you're in a `USE_INTERRUPTS` block, and you can reorder functions as needed without accidentally inserting something into the `USE_INTERRUPTS` block that you didn't intend to be interrupt-specific.